app/code/Magento/Quote/Model/Quote.php

```diff
@@ -2298,6 +2298,8 @@ public function validateMinimumAmount($multishipping = false)
             $storeId
         );
 
+        $this->collectTotals()->save();
```

Comment:
I'm not sure that triggering recollect totals in this method is good solution.
I believe that we can extend `QuoteRepository` with additional param `triggerRecollect` that will set `triggerRecollect` data at `Quote` object.
https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Quote/Model/Quote.php#L2461-L2470

@ihor-sviziev what do you think?

Reply:
@viniciusbord9 Could you pay attention to this comment? Extending of `QuoteRepository` allow to reuse recollect functional in a future.
Or may be you have another points for this case?

Reply:
@Den4ik 

I was waiting for an answer from @ihor-sviziev to whom you asked for sooner. I am going to try to implement your idea and check if it is viable.

Reply:
@Den4ik I think it's a good idea

Reply:
@Den4ik @ihor-sviziev 

I was developing the new repository extending the `QuoteRepository` as suggested and due to a necessity of injecting the repository for two classes(`TotalsInformationManagement`, `CartTotalRepository`) , I have realized that maybe the current flow was not working well.

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Checkout/Model/TotalsInformationManagement.php

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Quote/Model/Cart/CartTotalRepository.php

The class `TotalsInformationManagement` depends on the `CartTotalRepository` and apply changes to the quote but do not save the data

https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Checkout/Model/TotalsInformationManagement.php#L48-L61

I have only change the line 61 from `$quote->collectTotals();` to `$quote->collectTotals()->save();` and then my changes worked as expected.

I have finished the implementation of the `QuoteRepository` supporting an additional parameter `$triggerRecollect` with default value 0. If you still think that change is the best one I will proceed, but to be honest I think it is too big that fix.